run_ci.sh

```diff
@@ -169,3 +169,7 @@ for FILE in $FILESTOCACHE ; do
         sudo cp -p $FILE $cached
     fi
 done
+
+# Now that the deployment is up, replace the machine-api-operator image with
+# one that needs to be tested in this CI.
+./run-custom-mao.sh quay.io/sdasu/machine-api-operator:sdasu-fix
```

Comment:
Can you put this part in a separate commit?  We'd drop this part before merging the rest.

Reply:
Yes, will do right after CI completes showing that the MAO commit linked here, works.

Reply:
@russellb CI has completed successfully. Is this a good time to remove the changes made to run_ci.sh?

Reply:
@sadasu Is anything doing validation to ensure that the new custom MAO comes up correctly and is operational?  It doesn't look like it ...

A suggestion:  Move where this runs to script 11 (register hosts), and run it before the worker machine set is scaled up from 0.  You can have it only run if some config variable is set pointing to a custom repo/branch of the MAO to run.  If you do this, when CI does its validation to ensure a worker comes up, you know that the worker only came up using the restarted custom MAO.